src/network/NetworkController.ts

```diff
@@ -78,36 +68,125 @@ export interface NetworkConfig extends BaseConfig {
  * @property isCustomNetwork - Identifies if the network is a custom network
  * @property provider - RPC URL and network name provider settings
  */
-export interface NetworkState extends BaseState {
+export type NetworkState = {
   network: string;
-  isCustomNetwork: boolean;
+  isCustomNetwork?: boolean;
   provider: ProviderConfig;
-  properties: NetworkProperties;
-}
+  properties?: NetworkProperties;
+};
 
 const LOCALHOST_RPC_URL = 'http://localhost:8545';
 
+const name = 'NetworkController';
+
+export type NetworkControllerStateChangeEvent = {
+  type: `NetworkController:stateChange`;
+  payload: [NetworkState, Patch[]];
+};
+
+export type NetworkControllerProviderChangeEvent = {
+  type: `NetworkController:providerChange`;
+  payload: [ProviderConfig]; // 'any' is actually 'new NetworkController().provider'
+};
+
+export type NetworkControllerGetProviderConfig = {
```

Comment:
What do you think about adding "Action" to the end of this? Same for `NetworkControllerGetEthQuery` below.